### ISSUE TITLE
Unable to login or test login to Danbooru

### ISSUE THREAD

**Issue by Lil-Sassy** [Bionus/imgbrd-grabber#2002]
**Bug description**

When attempting to Login via the Source menu and Danbooru Options, I get either a failure or unable to test. Sometimes the test button will disappear entirely unless I exit out of the menu and then go back in.

**Steps to reproduce**

1. Go to 'Sources -> Danbooru Options -> Login'
2. Enter Username & Password or API key

**Expected behavior**

I expect a Success

**System information**

- OS: Windows 10
- Grabber version: 7.3.0

**Additional context**
I'm sure that the password and API key that I'm entering are correct. Able to login to Danbooru just fine. Using a free account.
I've also tried the login with other image board options and they work fine.

**Comment by FichteFoll** [comment]
Can confirm in Arch Linux (and some commits before 7.3.0). This used to work with the same credentials a while back.

**Comment by xodiack** [comment]
I have/had this issue previously that resolved itself via the nightly and appears to occur again with 7.3.0. I was able to get Danbooru to login through grabber by doing the following:

1. Deselect all sources, including Danbooru
2. Exit and reopen Grabber
3. Searched a paywalled tag (to make sure it was actually logging in, not sure if this step would be neccessary.) This is without any sources selected.
4. Finally I re-enabled danbooru as a source, repeated my search and got results.

I don't know why this would work, but hopefully it could give some help on tracking down what is creating the issue. Another thing to add, is that I did also create a new API key when I first had this issue previously, but in the case of when I updated to 7.3.0 I did not.

**Comment by AyeMauro** [comment]
> 
> 
> I have/had this issue previously that resolved itself via the nightly and appears to occur again with 7.3.0. I was able to get Danbooru to login through grabber by doing the following:
> 
>     1. Deselect all sources, including Danbooru
> 
>     2. Exit and reopen Grabber
> 
>     3. Searched a paywalled tag (to make sure it was actually logging in, not sure if this step would be neccessary.) This is without any sources selected.
> 
>     4. Finally I re-enabled danbooru as a source, repeated my search and got results.
> 
> 
> I don't know why this would work, but hopefully it could give some help on tracking down what is creating the issue. Another thing to add, is that I did also create a new API key when I first had this issue previously, but in the case of when I updated to 7.3.0 I did not.

I attempted this a few times, but it didn't seem to work for me.

**Comment by xodiack** [comment]
The only other thing I forgot to include is I had a valid login for my username+password as well as the username+api key placed. Though I'm not sure if grabber keeps both when changing the settings or only the most recently selected one.

 Looking at it now, when I open it for the first time today the login portion is set to Http Basic with the API login.

**Comment by Penguin-Guru** [comment]
I seem to be having the same issue, but can't say for sure because this is my first time using the program. The default when I opened the window was HTTP Basic. Since I'm pretty sure that's not what I want even if it does work, I tried the other two. Through URL seems like it should work if the program is scraping HTML but I don't know if it would work with their API. POST should work for the API but I don't know if it works for HTML requests. That might be like trying to log in again every time you load a page, assuming they handle it.
